internal/config/identity/openid/jwt.go

```diff
@@ -526,7 +545,35 @@ func LookupConfig(kvs config.KVS, transport *http.Transport, closeRespFn func(io
 		c.DiscoveryDoc.ScopesSupported = scopes
 	}
 
+	if c.RolePolicy != "" {
+		// RolePolicy is valided by IAM System during its
+		// initialization.
+
+		// Generate role ARN as combination of provider domain and
+		// prefix of client ID.
+		domain := configURLDomain
+		if domain == "" {
+			// Attempt to parse the JWKs URI.
+			domain, _, _ = net.SplitHostPort(c.JWKS.URL.Host)
+			if domain == "" {
+				return c, config.Errorf("unable to generate a domain from the OpenID config.")
+			}
+		}
+		clientIDFragment := c.ClientID[:8]
+		if clientIDFragment == "" {
+			return c, config.Errorf("unable to get a non-empty clientID fragment from the OpenID config.")
+		}
+		resourceID := domain + "_" + clientIDFragment
+		c.roleArn, err = arn.NewIAMRoleARN(resourceID)
+		if err != nil {
+			return c, config.Errorf("unable to generate ARN from the OpenID config: %v", err)
+		}
+	}
+
 	if c.ClaimName == "" {
+		if c.RolePolicy != "" {
+			return c, config.Errorf("Claim Name cannot be set with Role Policy")
+		}
```

Comment:
Since we allow it why bother erroring out here? - basically RolePolicy overrides this value anyways.

Reply:
If the user is setting Claim Name explicitly along with role policy, they have some misunderstanding of how it works. It makes sense to error out for this case. It should be either via claim name or via role policy, not both.

Reply:
Yeah makes sense.